charts/drupal/templates/_helpers.tpl

```diff
@@ -373,23 +373,21 @@ done
 {{- define "drupal.extract-reference-data" -}}
 set -e
 if [[ "$(drush status --fields=bootstrap)" = *'Successful'* ]] ; then
-  # Figure out which tables to skip.
-  IGNORE_TABLES=""
-  IGNORED_TABLES=""
-  for TABLE in `drush sql-query "show tables;" | grep -E '{{ .Values.referenceData.ignoreTableContent }}'` ;
+  echo "Dump reference database."
+  mkdir /tmp/reference-data/
+  for table in $(drush sql-query 'show tables;');
   do
-    IGNORE_TABLES="$IGNORE_TABLES --ignore-table=$DB_NAME.$TABLE";
-    IGNORED_TABLES="$IGNORED_TABLES $TABLE";
+    # Add --no-data parameter for tables that match .Values.referenceData.ignoreTableContent.
+    [[ "${table}" =~ {{ .Values.referenceData.ignoreTableContent }} ]] && nodata='--no-data' || nodata=''
+    echo "Dumping table: ${table}"
+    mysqldump --user="${DB_USER}" --password="${DB_PASS}" --host="${DB_HOST}" "${nodata}" "${DB_NAME}" "${table}" > /tmp/reference-data/"${table}".sql
   done
 
-  echo "Dump reference database."
-  mysqldump -u $DB_USER --password=$DB_PASS --host=$DB_HOST $IGNORE_TABLES $DB_NAME > /tmp/db.sql
-  mysqldump -u $DB_USER --password=$DB_PASS --host=$DB_HOST --no-data $DB_NAME $IGNORED_TABLES >> /tmp/db.sql
-
-  # Compress the database dump and copy it into the backup folder.
+  # Compress the sql files into a single file and copy it into the backup folder.
   # We don't do this directly on the volume mount to avoid sending the uncompressed dump across the network.
-  gzip -1 /tmp/db.sql
-  cp /tmp/db.sql.gz /app/reference-data/db.sql.gz
+  cd /tmp/reference-data/ || exit
+  tar -cf /tmp/db.tar.gz -I 'gzip -1' ./*.sql
```

Comment:
check out `tar -C`, cd becomes irrelevant on the previous line.
https://linux.die.net/man/1/tar

aha, I see you are already using it below while untarring.
maybe you needed explicit exit when the dir is not there?

Reply:
I wanted to make sure only *.sql files are being included in the archive, and ran into this problem when trying to use -C option: https://superuser.com/questions/266422/tar-c-with-a-wildcard-file-pattern

Reply:
aha, i didnt know!